### ISSUE TITLE
Per-cluster biases in AdaptiveLogSoftmaxWithLoss

### ISSUE THREAD

**Issue by TevenLeScao** [pytorch/pytorch#37155]
## 🚀 Feature
Add per-cluster biases in addition to the main head bias in Pytorch's implementation of adaptive softmax.

## Motivation

[TransformerXL](https://arxiv.org/abs/1901.02860)'s implementation of adaptive softmax incorporates a bias for all the clusters. It would be nice to have Pytorch at parity with that SOTA.

## Pitch

I would like to add a flag to the class constructor to add per-cluster biases, and then use them in the logprob calculations.

## Additional context

Hugging Face already uses code by @thomwolf that only deviates from Pytorch code in this regard. I can quickly fix up a PR that adds the possibility of cluster biases if this feature is deemed interesting.

**Comment by gchanan** [comment]
@zhangguanheng66: do you think this is worth incorporating in Transformers?  If so, is the loss used outside of Transformers or should it be limited to that scope?
